ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityAdapter.java

```diff
@@ -57,92 +56,64 @@
-
     /**
      * An Android {@link Activity} weak reference used to show ads.
      */
     private WeakReference<Activity> mActivityWeakReference;
 
     /**
-     * Unity adapter delegate to to forward the events from {@link UnitySingleton} to Google Mobile
-     * Ads SDK.
+     * UnityBannerAd instance.
      */
-    private final UnityAdapterDelegate mUnityAdapterDelegate = new UnityAdapterDelegate() {
-
-        @Override
-        public String getPlacementId() {
-            return mPlacementId;
-        }
+    private UnityBannerAd bannerAd;
 
+    /**
+     * IUnityAdsLoadListener instance.
+     */
+    private IUnityAdsLoadListener mUnityLoadListener = new IUnityAdsLoadListener() {
         @Override
-        public void onUnityAdsReady(String placementId) {
-            // Unity Ads is ready to show ads for the given placementId. Send Ad Loaded event if the
-            // adapter is currently loading ads.
-            if (placementId.equals(getPlacementId()) && mMediationInterstitialListener != null) {
+        public void onUnityAdsAdLoaded(String s) {
+            Log.d(TAG, "Unity Ads interstitial ad successfully loaded for placement ID '"
+                    + s + "'.");
+            if (mMediationInterstitialListener != null) {
                 mMediationInterstitialListener.onAdLoaded(UnityAdapter.this);
             }
         }
 
         @Override
-        public void onUnityAdsStart(String placementId) {
-            // Unity Ads video ad started playing. Google Mobile Ads SDK does not support
-            // callbacks for Interstitial ads when they start playing.
-        }
-
-        @Override
-        public void onUnityAdsClick(String s) {
-            // Unity Ads ad clicked.
+        public void onUnityAdsFailedToLoad(String s) {
+            Log.e(TAG, "Unity Ads interstitial ad load failure for placement ID '"
+                    + getPlacementId() + "' " + s);
             if (mMediationInterstitialListener != null) {
-                mMediationInterstitialListener.onAdClicked(UnityAdapter.this);
-                // Unity Ads doesn't provide a "leaving application" event, so assuming that the
-                // user is leaving the application when a click is received, forwarding an on ad
-                // left application event.
-                mMediationInterstitialListener.onAdLeftApplication(UnityAdapter.this);
+                if (UnityAds.getPlacementState(getPlacementId()) == UnityAds.PlacementState.NO_FILL ||
+                        UnityAds.getPlacementState(getPlacementId()) == UnityAds.PlacementState.DISABLED) {
+                    mMediationInterstitialListener.onAdFailedToLoad(UnityAdapter.this,
+                            AdRequest.ERROR_CODE_NO_FILL);
+                }
+                else {
+                    mMediationInterstitialListener.onAdFailedToLoad(UnityAdapter.this,
+                            AdRequest.ERROR_CODE_INTERNAL_ERROR);
+                }
```

Comment:
Is it a copy paste from previous implementation?

what about simply calling `mMediationInterstitialListener.onAdFailedToLoad(UnityAdapter.this, AdRequest.ERROR_CODE_NO_FILL);`?

Is there any reason to check `UnityAds.getPlacementState(getPlacementId())`?

Reply:
Yeah i saw it somewhere else, I thought it would add specificity if we want to distinguish between a no fill or another type of internal error?

Reply:
at the moment we have no way to distinguish... `onUnityAdsFailedToLoad` == no fill at the moment.